### ISSUE TITLE
Can support ImGuiConfigFlags_ViewportsEnable?

### ISSUE THREAD

**Issue by Wuqiqi123** [dpaulat/imgui-backend-qt#1]
I test it in qt5 and it works perfect. 
Can you support Viewports and docking, I really like this two features.
Thanks.

**Comment by dpaulat** [comment]
Excellent - great to hear it works with Qt 5!

The backend is designed to work with Qt graphics contexts defined external to ImGui. With multi-viewports, ImGui manages the viewports and graphics contexts. I played around with it briefly, but I wasn't able to find a good way to separate that functionality and make them externally managed. The solution I had was to create an ImGui context for each widget, and that seemed to work well for my purposes. That said, I'm open to alternate solutions.

I think docking might be a little bit easier to achieve. First, instead of creating an ImGui context per widget, you'd have to return to sharing a single ImGui context (this should work decently, I've tested this part). In addition, you'll need to enable docking via the enable flag using the docking branch. Per the [Docking](https://github.com/ocornut/imgui/wiki/Docking) documentation, that should get you 90% of the features. While it's a starting point, I'm skeptical, as I haven't tested it with this backend. I do have some code that checks whether input occurred on a given widget, and I suspect this might need some tweaking.

I'm not sure what an entire-screen docking approach would look like, I would imagine there's a way to create an invisible Qt OpenGL widget over the entire screen as your main viewport.

There has been at least some discussion around a similar use case:
https://github.com/ocornut/imgui/issues/4298

**Comment by Wuqiqi123** [comment]
Yes! I checkout imgui to docking branch and test DemoWindow. It looks works.
![Peek 2023-01-13 13-50](https://user-images.githubusercontent.com/37368540/212247243-bcb7b95f-d4ae-415a-857f-b5b5d9f9dbc8.gif)
